instrumentation/spring/spring-ws-2.0/javaagent/src/test/groovy/test/boot/SpringWsTest.groovy

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test.boot
+
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import io.opentelemetry.instrumentation.test.asserts.TraceAssert
+import io.opentelemetry.instrumentation.test.base.HttpServerTestTrait
+import io.opentelemetry.sdk.trace.data.SpanData
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import io.opentelemetry.test.hello_web_service.HelloRequest
+import io.opentelemetry.test.hello_web_service.HelloRequestSoapAction
+import io.opentelemetry.test.hello_web_service.HelloRequestWsAction
+import io.opentelemetry.test.hello_web_service.HelloResponse
+import org.springframework.boot.SpringApplication
+import org.springframework.context.ConfigurableApplicationContext
+import org.springframework.oxm.jaxb.Jaxb2Marshaller
+import org.springframework.util.ClassUtils
+import org.springframework.ws.client.core.WebServiceMessageCallback
+import org.springframework.ws.client.core.WebServiceTemplate
+import org.springframework.ws.soap.addressing.client.ActionCallback
+import org.springframework.ws.soap.client.SoapFaultClientException
+import org.springframework.ws.soap.client.core.SoapActionCallback
+import spock.lang.Shared
+
+class SpringWsTest extends AgentTestRunner implements HttpServerTestTrait<ConfigurableApplicationContext> {
```

Comment:
Have you thought about extending `HttpServerTest`? Some of our frameworks instrumentations use it for tests

Reply:
Imo `HttpServerTest` is not suitable for testing a soap webservice. To use `HttpServerTest` I would probably need to disable most of the test and hard code soap request content instead of using `WebServiceTemplate`.